### PR TITLE
New PR

### PR DESCRIPTION
**Pull request by raj-bhagat**

### CODE REVIEW

**Review by kanaksony — DISMISSED** [review]
Good to merge as all sonar issues are fixed.

File: entry/src/main/java/com/example/appcrashtracker/slice/MainAbilitySlice.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    @Override
+    public void onStart(Intent intent) {
+        super.onStart(intent);
+        super.setUIContent(ResourceTable.Layout_ability_main);
+        Button bt = (Button) findComponentById(ResourceTable.Id_crashButton);
+
+        bt.setClickedListener(component -> crashme());
+    }
+
+    private void crashme() {
```
Comment (kanaksony) [inline entry/src/main/java/com/example/appcrashtracker/slice/MainAbilitySlice.java:23]: Please follow the camel case convention in naming.

File: entry/src/main/config.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        "orientation": "unspecified",
+        "name": "com.example.appcrashtracker.MainAbility",
+        "icon": "$media:icon",
+        "description": "$string:mainability_description",
+        "label": "$string:entry_MainAbility",
+        "type": "page",
+        "launchType": "standard"
+      },
+      {
+        "orientation": "unspecified",
```
Comment (kanaksony) [inline entry/src/main/config.json:64]: I don't see this ability being used anywhere, please remove it.

File: entry/src/main/java/com/example/appcrashtracker/MainAbility2.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.example.appcrashtracker;
+
+import ohos.aafwk.ability.Ability;
+import ohos.aafwk.content.Intent;
+import com.example.appcrashtracker.slice.MainAbility2Slice;
+
+/** This main abilitly2 created for checking the library. */
+public class MainAbility2 extends Ability {
```
Comment (kanaksony) [inline entry/src/main/java/com/example/appcrashtracker/MainAbility2.java:8]: I don't see this ability slice being used anywhere, please delete it.

File: appcrashtracker/src/main/java/com/org/appcrashtracker/ExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+    private boolean externalFreeSpace = false;
+    private boolean packageName = false;
+    private boolean networkMode = false;
+    private boolean country = false;
+
+    public ExceptionHandler(Ability ability) throws NotExistException, WrongTypeException, IOException {
+        this.ability = ability;
+        abilityName = ability.getClass().getSimpleName();
+        ResourceManager resources = ability.getResourceManager();
+        String postUrl = ability.getString(String_url);
```
Comment (kanaksony) [inline appcrashtracker/src/main/java/com/org/appcrashtracker/ExceptionHandler.java:69]: This declaration and assignment can be removed and directly used in next line.

File: appcrashtracker/src/main/java/com/org/appcrashtracker/ExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+        if (postUrl.equals("default_url"))
+            HiLog.error(label, "Post url not specified");
+        else {
+            className = resources.getElement(Boolean_class_name).getBoolean();
+            message = resources.getElement(Boolean_message).getBoolean();
+            localizedMessage = resources.getElement(Boolean_localized_message).getBoolean();
+            causes = resources.getElement(Boolean_causes).getBoolean();
+            stackTraceBoolean = resources.getElement(Boolean_stack_trace).getBoolean();
+            brandName = resources.getElement(Boolean_brand_name).getBoolean();
```
Comment (kanaksony) [inline appcrashtracker/src/main/java/com/org/appcrashtracker/ExceptionHandler.java:78]: We can create a data class for all these values wrapped inside and populate that class object and then create JSON out of it.

**Review by kanaksony — CHANGES_REQUESTED** [review]
Please resolve the comments.

File: entry/src/main/java/com/example/appcrashtracker/slice/MainAbilitySlice.java (hunk 1 above)
Comment (raj-bhagat) [reply entry/src/main/java/com/example/appcrashtracker/slice/MainAbilitySlice.java]: Resolved in commit 00bd46e77851473750360a5a91d148105bf9950c

File: entry/src/main/config.json (hunk 2 above)
Comment (raj-bhagat) [reply entry/src/main/config.json]: Resolved in commit 00bd46e77851473750360a5a91d148105bf9950c

File: entry/src/main/java/com/example/appcrashtracker/MainAbility2.java (hunk 3 above)
Comment (raj-bhagat) [reply entry/src/main/java/com/example/appcrashtracker/MainAbility2.java]: Resolved in commit 00bd46e77851473750360a5a91d148105bf9950c

File: appcrashtracker/src/main/java/com/org/appcrashtracker/ExceptionHandler.java (hunk 4 above)
Comment (raj-bhagat) [reply appcrashtracker/src/main/java/com/org/appcrashtracker/ExceptionHandler.java]: Resolved in commit 00bd46e77851473750360a5a91d148105bf9950c

File: appcrashtracker/src/main/java/com/org/appcrashtracker/ExceptionHandler.java (hunk 5 above)
Comment (raj-bhagat) [reply appcrashtracker/src/main/java/com/org/appcrashtracker/ExceptionHandler.java]: Resolved in commit 00bd46e77851473750360a5a91d148105bf9950c

File: AppCrashTracker/src/main/java/com/org/appcrashtracker/data/ExceptionData.java [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+    private boolean externalMemorySize = false;
+    private boolean internalFreeSpace = false;
+    private boolean externalFreeSpace = false;
+    private boolean packageName = false;
+    private boolean networkMode = false;
+    private boolean country = false;
+    private ZSONObject jsonObjectData = new ZSONObject();
+
+    /** Set the conditions of the details required */
+    public void setBoolean(Ability ability) throws NotExistException, WrongTypeException, IOException {
```
Comment (kanaksony) [inline AppCrashTracker/src/main/java/com/org/appcrashtracker/data/ExceptionData.java:62]: No need for this method setBoolean. Booleans should be set in the constructor of ExceptionData class. Also, add Copyright text from the community copyright template.

File: appcrashtracker/src/main/java/com/org/appcrashtracker/ExceptionHandler.java [hunk 7]
```diff
@@ -1,124 +1,45 @@
...
-    private boolean batteryChargingVia = false;
-    private boolean sdCardStatus = false;
-    private boolean internalMemorySize = false;
-    private boolean externalMemorySize = false;
-    private boolean internalFreeSpace = false;
-    private boolean externalFreeSpace = false;
-    private boolean packageName = false;
-    private boolean networkMode = false;
-    private boolean country = false;
+    ExceptionData data = new ExceptionData();
```
Comment (kanaksony) [inline appcrashtracker/src/main/java/com/org/appcrashtracker/ExceptionHandler.java:22]: Should have private access modifier and only variable declaration should be there. Initialization must be in Constructor at L#67.

File: appcrashtracker/src/main/java/com/org/appcrashtracker/ExceptionHandler.java [hunk 8]
```diff
@@ -1,124 +1,45 @@
...
-            sdCardStatus = resources.getElement(Boolean_sd_card_status).getBoolean();
-            internalMemorySize = resources.getElement(Boolean_internal_memory_size).getBoolean();
-            externalMemorySize = resources.getElement(Boolean_external_memory_size).getBoolean();
-            internalFreeSpace = resources.getElement(Boolean_internal_free_space).getBoolean();
-            externalFreeSpace = resources.getElement(Boolean_external_free_space).getBoolean();
-            packageName = resources.getElement(Boolean_package_name).getBoolean();
-            networkMode = resources.getElement(Boolean_network_mode).getBoolean();
-            country = resources.getElement(Boolean_country).getBoolean();
-        }
+        data.setBoolean(ability);
```
Comment (kanaksony) [inline appcrashtracker/src/main/java/com/org/appcrashtracker/ExceptionHandler.java:26]: No need of this.

**Review by kanaksony — CHANGES_REQUESTED** [review]

**Review by kanaksony — APPROVED** [review]
Looks good.

**Review by sharathks84 — APPROVED** [review]
OK approved

**Review by Gleelakrishna — APPROVED** [review]
ok

**Pull request merged**